cluster-autoscaler/cloudprovider/oci/nodepools/oci_manager.go

```diff
@@ -0,0 +1,678 @@
+			}
+			switch item.LifecycleState {
+			case core.InstanceLifecycleStateStopped, core.InstanceLifecycleStateTerminated:
+				klog.V(4).Infof("skipping instance is in stopped/terminated state: %q", *item.Id)
+			case core.InstanceLifecycleStateCreatingImage, core.InstanceLifecycleStateStarting, core.InstanceLifecycleStateProvisioning, core.InstanceLifecycleStateMoving:
+				instances = append(instances, cloudprovider.Instance{
+					Id: *item.Id,
+					Status: &cloudprovider.InstanceStatus{
+						State: cloudprovider.InstanceCreating,
+					},
+				})
+			// in case an instance is running, it could either be installing OKE software or become a Ready node.
+			// we do not know, but as we only need info if a node is stopped / terminated, we do not care
+			case core.InstanceLifecycleStateRunning:
+				instances = append(instances, cloudprovider.Instance{
+					Id: *item.Id,
+					Status: &cloudprovider.InstanceStatus{
+						State: cloudprovider.InstanceRunning,
+					},
+				})
+			case core.InstanceLifecycleStateStopping, core.InstanceLifecycleStateTerminating:
+				instances = append(instances, cloudprovider.Instance{
+					Id: *item.Id,
+					Status: &cloudprovider.InstanceStatus{
+						State: cloudprovider.InstanceDeleting,
+					},
+				})
+			default:
+				klog.Warningf("instance found in unhandled state: (%q = %v)", *item.Id, item.LifecycleState)
+			}
+		}
+
+		// pagination logic
+		if r.OpcNextPage != nil {
+			// if there are more items in next page, fetch items from next page
+			request.Page = r.OpcNextPage
+		} else {
+			// no more result, break the loop
+			break
+		}
+	}
+
+	return len(instances), nil
+}
+
+func getDisplayNamePrefix(clusterId string, nodePoolId string) string {
+	shortNodePoolId := nodePoolId[len(nodePoolId)-11:]
+	shortClusterId := clusterId[len(clusterId)-11:]
+	return "oke" +
+		"-" + shortClusterId +
+		"-" + shortNodePoolId
+}
+
+// GetNodePoolNodes returns NodePool nodes that are not in a terminal state.
+func (m *ociManagerImpl) GetNodePoolNodes(np NodePool) ([]cloudprovider.Instance, error) {
+	klog.V(4).Infof("getting nodes for node pool: %q", np.Id())
+
+	nodePool, err := m.nodePoolCache.get(np.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	var instances []cloudprovider.Instance
+	for _, node := range nodePool.Nodes {
+
+		if node.NodeError != nil {
+
+			errorClass := cloudprovider.OtherErrorClass
+			if *node.NodeError.Code == "LimitExceeded" ||
+				(*node.NodeError.Code == "InternalServerError" &&
+					strings.Contains(*node.NodeError.Message, "quota")) {
+				errorClass = cloudprovider.OutOfResourcesErrorClass
+			}
+
+			instances = append(instances, cloudprovider.Instance{
+				Id: *node.Id,
+				Status: &cloudprovider.InstanceStatus{
+					ErrorInfo: &cloudprovider.InstanceErrorInfo{
+						ErrorClass:   errorClass,
+						ErrorCode:    *node.NodeError.Code,
+						ErrorMessage: *node.NodeError.Message,
+					},
+				},
+			})
+
+			continue
+		}
+
+		switch node.LifecycleState {
```

Comment:
If you could add the following line to the instance-pool implementation (`oci_instance_pool_manager.go` `GetInstancePoolNodes`) the behaviors would be consistent: 

```
@@ -261,6 +264,8 @@ func (m *InstancePoolManagerImpl) GetInstancePoolNodes(ip InstancePoolNodeGroup)
        for _, instance := range *instanceSummaries {
                status := &cloudprovider.InstanceStatus{}
                switch *instance.State {
+               case string(core.InstanceLifecycleStateStopped), string(core.InstanceLifecycleStateTerminated):
+                       klog.V(4).Infof("skipping instance is in stopped/terminated state: %q", *instance.Id)
                case string(core.InstanceLifecycleStateRunning):
                        status.State = cloudprovider.InstanceRunning
                case string(core.InstanceLifecycleStateCreatingImage):
```